### ISSUE TITLE
Startup issue - matchit.vim loaded when using packages

### ISSUE THREAD

**Issue by gegoune** [andymass/vim-matchup#176]
### Explain the issue

Hey, matchup is trying to disable built in matchit plugin, doesn't it?

If matchup is loaded using `packages` (as `start/` plugin) I think it is setting `g.loaded_matchit = 1` too late and distributed match it gets loaded anyway.

I discovered it by seeing 
```
    074.072  000.237  000.237: sourcing /usr/local/Cellar/neovim/HEAD-3f526fe/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
    074.349  004.363  004.126: sourcing /usr/local/Cellar/neovim/HEAD-3f526fe/share/nvim/runtime/plugin/matchit.vim
 ```
in my `--startuptime`

When I add `vim.g.loaded_matchit = 1` to my `init.lua` it takes way less to source that file indicating that it is catching the guard at the top and exiting early.

```
    069.794  000.028  000.028: sourcing /usr/local/Cellar/neovim/HEAD-3f526fe/share/nvim/runtime/plugin/matchit.vim
    070.089  000.208  000.208: sourcing /usr/local/Cellar/neovim/HEAD-3f526fe/share/nvim/runtime/plugin/matchparen.vim
``` 

It's not really a bug in matchup, hence I am not providing exact reproduction step - I am just making you aware of the problem with vim packages I guess. I know it'a only 4ms so not a big deal at all, it'a more about not working assumption.

### Minimal vimrc file

Just load the plugin with packer.

**Comment by andymass** [comment]
Thanks.  I am not sure if there is really anything I can do about it because it is neovim automatically loading matchit, and there is no way for the user to prevent it (except for `g.loaded_matchit = 1`).  Do you know of another workaround?

**Comment by gegoune** [comment]
Nope, no idea, was really hoping you will have one! :) Perhaps a little disclaimer in readme with regards to using vim packages? Not sure if there is anything negative with loading distributed script before matchup gets loaded. If there aren't any downsides then it probably doesn't matter. I for one will just add that var to my init.lua

**Comment by andymass** [comment]
Have to say, I am not too happy with the way that neovim forcibly loads matchit on users.  You could raise it as a bug on neovim but I am guessing they would likely say that the accepted fix is just `g.loaded_matchit = 1`.

I can confirm there should be no issue with load order on either vim or neovim.
It is very difficult to deal with these little differences between platforms, but this is one which I have explicitly considered and handled.  Of course, if neovim were to change matchit this could break things in an unpredictable way, so there's that risk.

I have some comments about this here but reading it now I think it's not really accurate for neovim, so I will update it https://github.com/andymass/vim-matchup#matchit
